static/js/reactions.js

```diff
@@ -142,15 +142,19 @@ exports.reaction_navigate = function (e, event_name) {
     var first_emoji = get_emoji_at_index(0);
     var selected_emoji = get_selected_emoji();
     var selected_index = find_index_for_emoji(selected_emoji);
-
+    var emoji_popover_filter = $('.emoji-popover-filter')[0];
     // special cases
     if (event_name === 'down_arrow') {
         if ($('.emoji-popover-filter').is(':focus') && first_emoji) { // move down into emoji map
             $(first_emoji).focus();
         }
     } else if (event_name === 'up_arrow') {
         if (selected_emoji && selected_index < 6) { // move up into reaction filter
-            $('.emoji-popover-filter').focus();
+            emoji_popover_filter.focus();
+            setTimeout(function () {
+                emoji_popover_filter.selectionStart = emoji_popover_filter.value.length;
+                emoji_popover_filter.selectionEnd =  emoji_popover_filter.value.length;
+            }, 0);
```

Comment:
caret function isn't supported by the base jquery.js. We will have to link the jquery.caret plugin to support caret function. And secondly, I tried to remove the cursor flickers, but the problem is, the focus event. The call to focus sets the cursor at the start as well. So, we have to set it after the cursor is focused, this creates the cursor flickers. I thought and searched through many things. But didn't find a satisfying solution.